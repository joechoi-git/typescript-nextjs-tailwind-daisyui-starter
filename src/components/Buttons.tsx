"use client";

import React from "react";
import Button from "./Button";

export default function Buttons(): React.JSX.Element {
    const [clicked, setClicked] = React.useState(0);
    const [message, setMessage] = React.useState("Clicked 0 time.");
    const handleClick = () => {
        setMessage(`Clicked ${clicked + 1} times.`);
        setClicked(clicked + 1);
    };
    return (
        <section className="p-4">
            <p>{message}</p>
            <div className="flex justify-center w-full py-2 gap-2">
                <Button onClick={handleClick}>button 1</Button>
                <Button onClick={handleClick}>button 2</Button>
                <Button onClick={handleClick}>button 3</Button>
            </div>
        </section>
    );
}
