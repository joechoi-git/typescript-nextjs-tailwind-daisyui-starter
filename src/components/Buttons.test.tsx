import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Buttons from "./Buttons";

// Mock Button component if it has complex functionality that doesn't need testing here
jest.mock("./Button", () =>
    // eslint-disable-next-line react/display-name
    ({ onClick, children }: { onClick: () => void; children: React.ReactNode }) => (
        <button onClick={onClick}>{children}</button>
    )
);

describe("Buttons Component", () => {
    test("renders initial state correctly", () => {
        render(<Buttons />);
        expect(screen.getByText("Clicked 0 time.")).toBeInTheDocument();
        expect(screen.getByText("button 1")).toBeInTheDocument();
        expect(screen.getByText("button 2")).toBeInTheDocument();
        expect(screen.getByText("button 3")).toBeInTheDocument();
    });

    test("updates message when buttons are clicked", () => {
        render(<Buttons />);

        const button1 = screen.getByText("button 1");
        const button2 = screen.getByText("button 2");
        const button3 = screen.getByText("button 3");

        // Click button 1
        fireEvent.click(button1);
        expect(screen.getByText("Clicked 1 times.")).toBeInTheDocument();

        // Click button 2
        fireEvent.click(button2);
        expect(screen.getByText("Clicked 2 times.")).toBeInTheDocument();

        // Click button 3
        fireEvent.click(button3);
        expect(screen.getByText("Clicked 3 times.")).toBeInTheDocument();
    });
});
