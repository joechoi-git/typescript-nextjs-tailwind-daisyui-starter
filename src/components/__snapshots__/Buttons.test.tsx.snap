// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Component matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="py-4"
  >
    <p>
      Clicked 0 time.
    </p>
    <div
      class="flex w-full py-2 gap-2"
    >
      <button>
        button 1
      </button>
      <button>
        button 2
      </button>
      <button>
        button 3
      </button>
    </div>
  </section>
</DocumentFragment>
`;
